### PR TITLE
Toggle reset button on option change

### PR DESCRIPTION
**Pull request by ssreerama**
This PR Adds the ability to toggle the reset button on option updates.
1. Enables when the option is changed.
2. Disables when the defaults are set.

Total scenarios: 
Operations - Reset state (Disable)
1. Opens options tab first time- disabled
2. Options changes - enable instantly
3. Options changes + ok - Enables
4. Options changes + cancel - Disable (or depends on prev state)
5. Options changed + reset clicked + cancel - Disable (or depends on prev state)
6. Options changed + reset clicked + ok - Disable
7. Reset clicked followed by ok clicked - Disable
8. Reset clicked +option changed + cancel- Disable (or depends on prev state)
9. Reset clicked + option changed +ok - Enable

When no option is changed:
image

When there is a change identified through the publish process:
image

### CODE REVIEW

File: extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts [hunk 1]
```diff
@@ -58,6 +58,7 @@ export class PublishDatabaseDialog {
 	private serverName: string | undefined;
 	protected optionsButton: azdataType.ButtonComponent | undefined;
 	private publishOptionsDialog: PublishOptionsDialog | undefined;
+	public publishOptionsModified: boolean = false;
```
Comment (kisantia) [inline extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts:61]: what if there were options specified in a publish profile that got loaded?

Comment (ssreerama) [reply extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts:61]: options from profile PR is not out yet!

But, we decide to consider the default options from profile if profile loaded (final design for profile under discussion) which means the Reset shouldn't be enabled while profile load right! as they can only reset to profile, toggle reset scenario will be same for profile options or default options!

correct me if I'm missing something! Thanks

Comment (kisantia) [reply extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts:61]: ok this is fine for now. Yes we still need to make a final decision on if reset should reset to the profile's loaded values or to the DacFx defaults

File: extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts [hunk 2]
```diff
@@ -145,10 +150,11 @@ export class PublishOptionsDialog {
 		// Set the publish deploymentoptions with the updated table component values
 		this.publish.setDeploymentOptions(this.optionsModel.deploymentOptions);
 		this.disposeListeners();
-
```
Comment (kisantia) [inline extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts:148]: nit: keep this empty line before the if for readability

File: extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts [hunk 3]
```diff
@@ -145,10 +150,11 @@ export class PublishOptionsDialog {
...
 		if (this.optionsChanged) {
 			TelemetryReporter.sendActionEvent(TelemetryViews.PublishOptionsDialog, TelemetryActions.optionsChanged);
 		}
+		// When options are Reset to default and clicked Ok, seting optionsChanged flag to false, if not set the state of the option change
+		this.publish.publishOptionsModified = this.isResetOptionsClicked ? false : (this.optionsChanged || this.publish.publishOptionsModified);
```
Comment (kisantia) [inline extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts:157]: what if the user changed an option or two after clicking reset?

File: extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts [hunk 4]
```diff
@@ -47,6 +48,8 @@ export class PublishOptionsDialog {
 
 		let resetButton = utils.getAzdataApi()!.window.createButton(constants.ResetButton);
 		resetButton.onClick(async () => await this.reset());
+		// If options values already modified then enable the reset button
+		resetButton.enabled = this.publish.publishOptionsModified!;
```
[the remark this answers is not in the document]
Comment (ssreerama) [reply extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts:52]: Actually not needed!

File: extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts (hunk 3 above)
Comment (ssreerama) [reply extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts]: Updated the total scenarios in the description, let me know if you think of any other scenarios. Thanks!

**Review by kisantia — APPROVED** [review]

**Pull request merged**